modules/reference/appvms/business.nix

```diff
@@ -11,6 +11,62 @@ let
   #TODO: Move this to a common place
   xdgPdfPort = 1200;
   name = "business";
+  
+  # use nix-prefetch-url to calculate sha256 checksum
+  endpointsFile = pkgs.fetchurl {
+    url = "https://endpoints.office.com/endpoints/worldwide?clientrequestid=b10c5ed1-bad1-445f-b386-b919946339a7";
+    sha256 = "1zly0g23vray4wg6fjxxdys6zzksbymlzggbg75jxqcf8g9j6xnw";
+  };
+
+  generateO365FWScript = pkgs.writeShellScript "generate-o365fw-script" ''
+    #!/usr/bin/env bash
+
+    set -euo pipefail
+
+    ENDPOINTS_FILE="${endpointsFile}"
+
+    preprocess_url() {
+        local url="$1"
+        # Return the URL as-is without any filtering
+        echo "$url"
+    }
+
+    generate_iptables_rules() {
+        echo "# Office 365 Firewall Rules"
+        echo
+
+        jq -r '.[] | select(.category == "Optimize" or .category == "Allow" or .category == "Default") | .ips[]?' "$ENDPOINTS_FILE" | sort -u | while read -r ip; do
+            if [[ $ip == *":"* ]]; then
+                echo "ip6tables -I OUTPUT -d $ip -j ACCEPT"
+                echo "ip6tables -I INPUT -d $ip -j ACCEPT"
+            else
+                echo "iptables -I OUTPUT -d $ip -j ACCEPT"
+                echo "iptables -I INPUT -d $ip -j ACCEPT"
+            fi
+        done
+
+        echo
+
+        jq -r '.[] | select(.category == "Optimize" or .category == "Allow" or .category == "Default") | .urls[]?' "$ENDPOINTS_FILE" | sort -u | while read -r url; do
+            processed_url=$(preprocess_url "$url")
+            if [[ "$processed_url" != "" ]]; then
+                echo "iptables -I OUTPUT -p tcp --dport 80 -m string --string \"$processed_url\" --algo bm -j ACCEPT"
+                echo "iptables -I INPUT -p tcp --dport 80 -m string --string \"$processed_url\" --algo bm -j ACCEPT"
+                echo "iptables -I OUTPUT -p tcp --dport 443 -m string --string \"$processed_url\" --algo bm -j ACCEPT"
+                echo "iptables -I INPUT -p tcp --dport 443 -m string --string \"$processed_url\" --algo bm -j ACCEPT"
```

Comment:
Same as previous comment. In addition, `--sport` should have been used rather than `--dport`.